src/hotspot/share/gc/shenandoah/mode/shenandoahSATBMode.cpp

```diff
@@ -60,7 +61,7 @@ ShenandoahHeuristics* ShenandoahSATBMode::initialize_heuristics() const {
   } else if (strcmp(ShenandoahGCHeuristics, "static") == 0) {
     return new ShenandoahStaticHeuristics();
   } else if (strcmp(ShenandoahGCHeuristics, "adaptive") == 0) {
-    return new ShenandoahAdaptiveHeuristics();
+    return new ShenandoahAdaptiveHeuristics(ShenandoahHeap::heap());
   } else if (strcmp(ShenandoahGCHeuristics, "compact") == 0) {
     return new ShenandoahCompactHeuristics();
```

Comment:
It feels a bit ad-hoc to me that ShenandoahStaticHeuristic & ShenandoahStaticHeuristic, which both use some information about the state of the heap (i.e. state queriable through ShenandoahHeapStats interface methods), don't get passed that argument in the constructor. I'd ideally like to see the c'tors changed to hew to the same pattern rather than their reaching directly into the global heap object to get what they need.

A question to ask is, if a new heuristic were to be designed in the future, would it extend ShenandahHeapStats, being the official interface through which heuristics make use of properties of the heap upon which they operate to extract that information?